internal/app/machined/pkg/system/services/kubelet.go

```diff
@@ -411,5 +412,32 @@ func pickNodeIPs(cidrs []string) ([]stdnet.IP, error) {
 		}
 	}
 
+	if len(result) > 1 {
```

Comment:
I do not think this is the right way, as we can't pick one IP if multiple IPs match (we have no idea which one is the "right" one). I would prefer kubelet to fail like it does today